beacon-chain/rpc/node_server.go

```diff
@@ -33,11 +33,13 @@ func (ns *NodeServer) GetSyncStatus(ctx context.Context, _ *ptypes.Empty) (*ethp
 
 // GetGenesis fetches genesis chain information of Ethereum 2.0.
 func (ns *NodeServer) GetGenesis(ctx context.Context, _ *ptypes.Empty) (*ethpb.Genesis, error) {
-	beaconState, err := ns.beaconDB.FinalizedState()
+	// TODO(3045): Use the db.Database interface only.
+	beaconState, err := ns.beaconDB.(*db.BeaconDB).FinalizedState()
```

Comment:
Does the interface need support for the finalized state? Also should we do a check if the type assertion works and throw error if not?

Reply:
This will panic if not the correct interface. It's not as nice of an error to check, but i didnt want to invest more time in careful checking for RPC since there is no other path right now anyway.